packages/connect-form/src/hooks/use-connection-color.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback } from 'react';
+
+type ColorCode =
+  | 'color1'
+  | 'color2'
+  | 'color3'
+  | 'color4'
+  | 'color5'
+  | 'color6'
+  | 'color7'
+  | 'color8'
+  | 'color9'
+  | 'color10';
```

Comment:
Obviously completely optional suggestion, but you can also do:

```suggestion
type ColorCode = `color${1|2|3|4|5|6|7|8|9|10}`;
```